flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/FlinkResourceValidator.java

```diff
@@ -31,4 +32,13 @@
      * @return Optional error string, should be present iff validation resulted in an error
      */
     Optional<String> validate(FlinkDeployment deployment);
+
+    /**
+     * Validate and return optional error.
+     *
+     * @param sessionJob the session job to be validated.
+     * @param session the target session cluster of the session job to be validated.
+     * @return Optional error string, should be present iff validation resulted in an error
+     */
+    Optional<String> validate(FlinkSessionJob sessionJob, Optional<FlinkDeployment> session);
```

Comment:
I think this method should be called validateSessionJob

Reply:
fixed